src/main/java/filter/expression/Qualifier.java

```diff
@@ -136,23 +131,61 @@ public static boolean process(IModel model, FilterExpression expr, TurboIssue is
         return exprWithNormalQualifiers.isSatisfiedBy(model, issue, new MetaQualifierInfo(metaQualifiers));
     }
 
+    /**
+     * Get all milestones that can be aliased with the keyword "current".
+     */
+    private static List<TurboMilestone> getMilestonesWithAlias(IModel model, List<String> repoIds) {
+        List<TurboMilestone> milestones = new ArrayList<>();
+        // add milestones with due date first
+        milestones.addAll(model.getMilestones().stream()
+                .filter(ms -> ms.getDueDate().isPresent())
+                .filter(ms -> repoIds.contains(ms.getRepoId().toLowerCase()))
+                .sorted((a, b) -> getMilestoneDueDateComparator().compare(a, b))
+                .collect(Collectors.toList()));
+
+        // Special case: if there is only one open milestone, it should be included regardless of whether
+        // it has due date or not.
+        // In this case, if it is not included already (the open milestone does not
+        // have due date), append it to the milestone list.
+        List<TurboMilestone> openMilestones = getOpenMilestones(model, repoIds);
+        if (openMilestones.size() == 1) {
+            TurboMilestone openMilestone = openMilestones.get(0);
+            boolean hasDueDate = openMilestone.getDueDate().isPresent();
+            if (!hasDueDate) {
+                milestones.add(openMilestone);
+            }
+        }
+
+        return milestones;
+    }
+
+    private static List<TurboMilestone> getOpenMilestones(IModel model, List<String> repoIds) {
+        return model.getMilestones().stream()
+                .filter(ms -> ms.isOpen())
+                .filter(ms -> repoIds.contains(ms.getRepoId().toLowerCase()))
+                .collect(Collectors.toList());
+    }
+
     private static Optional<Integer> getCurrentMilestoneIndex(List<TurboMilestone> allMilestones) {
```

Comment:
A more cohesive method could be like this:
```java
/**
 * Returns a Map where the key indicates offset from 'current' milestone. 
 * i.e. 0 for the 'current' milestone and -1 for the 'current-1' milestone and so on.
 */
Map<Integer, TurboMilestone> addMilestoneAliasIndex(List<TurboMilestone> milestones)
```

1. This method makes sense on its own and doesn't depend on the call site to sort the input first.
2. If sorting is required within the method, it should be done by a static method in the `TurboMilestone` class?
1. I'm not sure if `Map` is the best data structure though. May be there is a better one.
2. The code should still clearly show the high-level logic (i.e. apply SLAP).
3. The header comment must describe the rules used for determining the index.